src/receipt.rs

```diff
@@ -136,7 +149,20 @@ impl Receipt {
         Self::Return { id, val, pc, is }
     }
 
-    pub const fn return_data(
+    pub fn return_data(
+        id: ContractId,
+        ptr: Word,
+        digest: Bytes32,
+        data: Vec<u8>,
+        pc: Word,
+        is: Word,
+    ) -> Self {
+        let len = bytes::padded_len(&data) as Word;
+
+        Self::return_data_with_len(id, ptr, len, digest, data, pc, is)
+    }
+
+    pub const fn return_data_with_len(
```

Comment:
Should we be publicly exposing this function that doesn't guarantee the length is padded correctly?

Reply:
This one will allow `const` directives and is used to restore raw serialization. It means the output will always be the representation of what is serialized, without making assumptions.